### PR TITLE
Add mobile OfflineStatus component

### PR DESCRIPTION
**Pull request by derekblank**
## What?
Adds a mobile component to display a visual indicator when working offline within the mobile editor. (Note: the functionality to subscribe to network status will be handled in a separate PR.) 

## Why?
Overall, to improve the user's experience when uploading media. 

Addresses:
* https://github.com/wordpress-mobile/gutenberg-mobile/issues/6405

## How?
Adds OfflineStatus component and associated icon/style files. 

## Testing Instructions
Applying the test patch will display the component within the editor when offline.

<details><summary>Test Diff</summary>

```diff
diff --git a/packages/block-editor/src/components/block-list/index.native.js b/packages/block-editor/src/components/block-list/index.native.js
index 810e23e4c1..19540eddb4 100644
--- a/packages/block-editor/src/components/block-list/index.native.js
+++ b/packages/block-editor/src/components/block-list/index.native.js
@@ -28,6 +28,7 @@ import {
 	DEFAULT_BLOCK_LIST_CONTEXT,
 } from './block-list-context';
 import { BlockDraggableWrapper } from '../block-draggable';
+import OfflineStatus from '../offline-status';
 import { useEditorWrapperStyles } from '../../hooks/use-editor-wrapper-styles';
 import { store as blockEditorStore } from '../../store';
 
@@ -235,6 +236,7 @@ export default function BlockList( {
 			onLayout={ onLayout }
 			testID="block-list-wrapper"
 		>
+			<OfflineStatus />
 			{ isRootList ? (
 				<BlockListProvider
 					value={ {
```

</details>

## Screenshots or screencast 
Light | Dark
-|-
Screenshot 2023-12-07 at 6 18 07 pm | Screenshot 2023-12-07 at 6 17 02 pm

### CODE REVIEW

File: packages/block-editor/src/components/offline-status/index.native.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		styles.text,
+		styles[ 'text--dark' ]
+	);
+
+	const iconStyle = usePreferredColorSchemeStyle(
+		styles.icon,
+		styles[ 'icon--dark' ]
+	);
+
+	return ! isConnected ? (
```
Comment (dcalhoun) [inline packages/block-editor/src/components/offline-status/index.native.js:38]: If you believe it to be most effective, we could go ahead and implement this testing diff shared in this PR. Adding a `__DEV__` check to this conditional would mean we would only need to remove the the `__DEV__` flag to ship this feature. 

Not necessarily requesting this be done, merely sharing it for thought.

File: packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java
Comment (dcalhoun) [inline packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/GutenbergBridgeJS2Parent.java:1]: Rebasing this branch atop the latest `feat/connectivity-status` should get rid of this file diff and the others related to the `isConnected` Hook. A new PR is opened in https://github.com/WordPress/gutenberg/pull/56861.

**Review by dcalhoun — COMMENTED** [review]
Overall, the changes look good and the status indicator correctly shows up when the network status changes. 🎉 

I noted the status appears invisible on Android when dark mode is enabled. Do you see this as well? 

<details><summary>Android Dark Mode Screenshot</summary>

android dark mode status invisible


</details> 

Also, I'm not sure if my eyes are playing tricks on me or if the status text and icon are _not_ vertically centered. WDYT? 

<details><summary>Status Vertically Centered</summary>

offline status not vertically centered


</details>

File: packages/block-editor/src/components/offline-status/index.native.js (hunk 1 above)
Comment (derekblank) [reply packages/block-editor/src/components/offline-status/index.native.js:38]: Updated to add the component behind a `__DEV__` flag, and rectified the commit history. 👍

**Comment by derekblank** [comment]
> I noted the status appears invisible on Android when dark mode is enabled. Do you see this as well?

I did not note this, but it appears like it may be theme-based as I noted the editor background is not dark in your screenshot. I can take a look, however. What theme were you using? Both examples in this screenshot are using [Quadrat](https://en-au.wordpress.org/themes/quadrat/):

Screenshot 2023-12-08 at 11 59 18 am

**Comment by derekblank** [comment]
> Also, I'm not sure if my eyes are playing tricks on me or if the status text and icon are _not_ vertically centered. WDYT?

I think it's the icon that is not correct. I used a placeholder icon I found for free, but was intending to replace it after we received some design feedback. I'll seek that out now and replace it here as we're leaving the PR open.

File: packages/block-editor/src/components/offline-status/index.native.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { Icon } from '@wordpress/components';
+import { offline as offlineIcon } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { useIsConnected } from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style';
```
Comment (dcalhoun) [inline packages/block-editor/src/components/offline-status/index.native.js:18]: Looks like my recent suggestion to drop the `native` suffix here resulted in test failures. It appears Metro correctly resolve the file when only a single, suffixed file is present. However, Jest fails resolution in that context. I recall encountering this before. We need to reinstate the `native` suffix here. 

Sorry to misguide you and for the inconvenience. ☹️

**Review by dcalhoun — COMMENTED** [review]
> > I noted the status appears invisible on Android when dark mode is enabled. Do you see this as well?
> 
> I did not note this, but it appears like it may be theme-based as I noted the editor background is not dark in your screenshot. I can take a look, however. What theme were you using? 

Interesting. My screenshot is from a site using [Geologist](https://wordpress.org/themes/geologist/). 

My screenshot showcases not only an invisible background, but invisible text. Even if the theme is at play and the editor background is incorrect (concerning unto itself), shouldn't the text still be visible? The offline status UI has its own background and text colors, correct?

**Pull request closed without merging**